### ISSUE TITLE
error message for `no_auto_create_user` differs from MySQL 5.7

### ISSUE THREAD

**Issue by XuHuaiyu** [pingcap/tidb#10931]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do?
Test the behavior of SQL Mode `NO_AUTO_CREATE_USER`.

2. What did you expect to see?
``` sql
mysql> grant all on * to 'not_exits'@'localhost';
ERROR 1133 (42000): Can't find any matching row in the user table
```

3. What did you see instead?
``` sql
tidb> grant all on * to 'not_exist'@'localhost';
ERROR 1105 (HY000): You are not allowed to create a user with GRANT
```

4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?
``` sql
Release Version: v3.0.0-rc.1-234-gc5a157a88
Git Commit Hash: c5a157a883a07417780664c1de97610cce84e78b
Git Branch: master
UTC Build Time: 2019-06-25 02:44:56
GoVersion: go version go1.12 darwin/amd64
Race Enabled: false
TiKV Min Version: 2.1.0-alpha.1-ff3dd160846b7d1aed9079c389fc188f7f5ea13e
Check Table Before Drop: false

```

**Comment by morgo** [comment]
TiDB is MySQL 8.0 compatible here, not 5.7 :(
```
mysql [localhost:5725] {root} ((none)) > grant all on *.* to 'not_exits';
ERROR 1133 (42000): Can't find any matching row in the user table

mysql [localhost:8016] {root} ((none)) > grant all on *.* to 'not_exits';
ERROR 1410 (42000): You are not allowed to create a user with GRANT
```
An explanation:
- MySQL 8.0 doesn't actually support the `NO_AUTO_CREATE_USER` sql mode. It will never allow this behavior (it's a security risk).
- The error code is not matching 8.0 though :(

**Comment by nullnotnil** [comment]
I am going to suggest closing this as won't fix, since the MySQL 8.0 error message is more informative. We can revisit if it proves to be a major compatibility problem. Thanks!
